### PR TITLE
chore: Add actions to scan code

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
repo config changes:
add github actions to scan code commits before pr is merged.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, release-0.7, release-1.0, release-1.1, release-1.23 ]
```
Comment (nilo19) [inline .github/workflows/codeql-analysis.yml:16]: can we make it to all branches instead of manually declare the name?

Comment (MartinForReal) [reply .github/workflows/codeql-analysis.yml]: change it to release-**

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, 'releases-**' ]
```
Comment (feiskyer) [inline .github/workflows/codeql-analysis.yml:16]: branch name should be `release-**` (no `s` at the end)

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master, 'releases-**' ]
```
Comment (feiskyer) [inline .github/workflows/codeql-analysis.yml:19]: same here

Comment (MartinForReal) [reply .github/workflows/codeql-analysis.yml]: updated

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (MartinForReal) [reply .github/workflows/codeql-analysis.yml]: updated

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, 'release-**' ]
```
Comment (feiskyer) [inline .github/workflows/codeql-analysis.yml:16]: could you add the same configure to azuredisk/azurefile CSI drivers as well?

Comment (MartinForReal) [reply .github/workflows/codeql-analysis.yml:16]: sure

**Pull request merged**